src/lib/log.c

```diff
@@ -38,14 +38,22 @@ static void log_internal(FILE *out, const char *file, int line, const char *labe
 	struct tm *timeinfo;
 	int printed;
 
-	/* if the message starts with '\n' add a line break before the message */
+	/* if the message starts with '\n' add a line break before the message
+	 * unless we are printing in debug mode, or to a file because that would
+	 * mess up the output, if that is the case remove those characters */
 	while (strncmp(format, "\n", 1) == 0 || strncmp(format, "\r", 1) == 0) {
-		if (out == stdout || out == stderr) {
+		if ((out == stdout || out == stderr) && (cur_log_level != LOG_DEBUG)) {
 			fprintf(out, "%c", format[0]);
 		}
 		format++;
 	}
 
+	/* if the message was empty or had only line breaks or carriage return
+	 * finish here, otherwise we could print empty debug info or labels */
+	if (strcmp(format, "") == 0) {
```

Comment:
Is this a real case? Are we using empty prints?

Reply:
We are not using empty strings, but with this implementation, if someone is printing to a file and the code has for example a `info("\n")`, since the \n is going to be stripped from the message, the resulting message would be an empty string.